tests/tests_firewalld_conf.yml

```diff
@@ -3,56 +3,53 @@
   hosts: all
   gather_facts: false
   tasks:
-    - name: Initial run, reset, disable zone drifting
+    - name: Start with a clean configuration
       include_role:
         name: linux-system-roles.firewall
       vars:
         firewall:
           previous: replaced
-          firewalld_conf:
-            allow_zone_drifting: "no"
-          permanent: true
 
-    - name: Assert AllowZoneDrifting is disabled
-      shell:
-        cmd: >-
-          set -o pipefail;
-          (grep AllowZoneDrifting /etc/firewalld/firewalld.conf
-          || echo "AllowZoneDrifting=no") | tail -1
+    - name: Determine if AllowZoneDrifting is deprecated
+      command: grep AllowZoneDrifting /etc/firewalld/firewalld.conf
       register: result
       changed_when: false
-      failed_when: result.stdout_lines[0] != "AllowZoneDrifting=no"
+      failed_when: result.rc == 2
+
+    - name: Set fact allow_zone_drifting_not_deprecated
+      set_fact:
+        allow_zone_drifting_not_deprecated: "{{ result.rc == 0 }}"
 
-    - name: Start with clean configuration
+    - name: Try to disable zone drifting
       include_role:
         name: linux-system-roles.firewall
       vars:
         firewall:
           previous: replaced
           firewalld_conf:
-            allow_zone_drifting: "yes"
+            allow_zone_drifting: no
           permanent: true
 
-    - name: Get firewalld version
-      command: firewall-cmd --version
-      register: result
-      changed_when: false
-
-    - name: Check if AllowZoneDrifting deprecated (>= v1.0.0)
-      set_fact:
-        allow_zone_drifting_deprecated: >-
-          "{{ result.stdout[0] | int > 0 | bool }}"
-
-    - name: Assert AllowZoneDrifting is enabled if possible
-      vars:
-        expected_result: >-
-          "{{ allow_zone_drifting_deprecated | ternary('no', 'yes') | quote }}"
+    - name: Check if AllowZoneDrifting is disabled if possible
       shell:
         cmd: >-
           set -o pipefail;
-          (grep AllowZoneDrifting /etc/firewalld/firewalld.conf
-          || echo "AllowZoneDrifting=no") | tail -1
+          grep AllowZoneDrifting /etc/firewalld/firewalld.conf | tail -1
```

Comment:
Do you want to check if `AllowZoneDrifting` is in the file but commented out?  e.g.
```
# AllowZoneDrifting=no
```
?

Reply:
This is changing, but I want to get the actual directive for AllowZoneDrifting.

Reply:
Changing this to use `grep -Fx AllowZoneDrifting=yes /etc/firewalld/firewalld.conf` instead.